app/src/main/java/com/android/voyageur/ui/formFields/UserDropdown.kt

```diff
@@ -27,24 +27,43 @@ import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
 import com.android.voyageur.model.user.User
 
+/**
+ * Composable function that displays the profile picture of a given user.
+ *
+ * This function renders a user's profile picture inside a circular avatar. If the user's profile
+ * picture URI is not available or is empty, it falls back to displaying the first letter of the
+ * user's name in uppercase, centered within the avatar circle.
+ *
+ * @param user An instance of the `User` class containing the user's profile data, including the
+ *   `profilePicture` URI and `name`. *
+ */
 @Composable
-fun UserIcon(text: String) {
+fun UserIcon(user: User) {
+  val profilePictureUri = user.profilePicture
   Box(
       modifier =
           Modifier.size(30.dp) // Set size for the avatar circle
               .testTag("participantAvatar")
               .background(Color.Gray, shape = RoundedCornerShape(50)), // Circular shape
       contentAlignment = Alignment.Center) {
-        Text(text = text.first().uppercaseChar().toString(), color = Color.White)
+        if (profilePictureUri != "") {
```

Comment:
Is there a case where is condition will be false? I am running the assumption that all users have a profile picture, even the ones that have it as the default google one.

So I think there's no need for the if and the text field on the else

Reply:
I thought about this.. and since our Sign In is only based on Google authentication you're right. I did this thinking of the case in which the user has their own account separate from Google (perhaps in the future) but it's true for the moment this isn't the case.

Reply:
Giving it another thought, wouldn't it make sense to have this, and in the future have the possibility of a user to delete their profile picture?

Reply:
If they want to delete a picture, they can just use a blank picture which they select. We don't have an option to delete the picture right now (and I don't think we will) so for now it doesn't make sense.

Reply:
I see your point and I think you're right. I'll remove the user icon with the initial.